### DISCUSSION TITLE
Reset Column Visibility of Datatable not working after changing primefaces 7 to 11

### DISCUSSION THREAD

**Discussion by AlexFTrapp** [primefaces/community#2920] [PrimeFaces]
Hi Forum,

i just upgraded my application from Primefaces 7 to primefaces 11. During that process i found that my logic to reset standard column visibility in my datatable by button is no longer working. I know that Primefaces11 is not the newest version existing, but the framework I use (Axon Ivy) comes with exactly that version and I cannot use another.
What the logic does:
Each column checks its visibility attribute from the lazyLoadDataModel which holds a Map with boolean values by columnIds. `<p:column visible="#{data.projectLazyDataModel.isColumnVisible(<ColumnID>)}"`. The ColumnToggler-Event changes the boolean values in that Map according to the user selection.
An additional button resets the Map in the LazyDataModel to default values (clearColumnSelection()) and then clears all existing filters and updates the table-layout (see below).
In Version 7 of Primefaces the table got updated that way and afterwards showed the original amount of columns. In version 11 of Primefaces the tablelayout is not updated. The resetFilter() is done correctly in both versions and I can see when debugging that the method to check the visibility in LazyDataModel is called for each column and it does return the correct boolean value, but still the columns remain visible.
Does anyone have any idea on how to change the xhtml in order to fix the logic?
Here a snippet of my button
`
<p:commandButton id="filterResetBtn"
	value="#{ivy.cms.co('/Labels/showDefault')}"
	styleClass="marginButton printhidden floatRight"
	style="width: fit-content;" update="projectTable"
	actionListener="#{data.projectLazyDataModel.clearColumnSelection()}"
	oncomplete="PF('projectTableWidget').clearFilters()">
	<p:resetInput target="projectTable" />
</p:commandButton>`

Many Thanks,
Alex

**Comment by melloware** [comment]
I don't think you want to clear like that anymore you can to call `dt.reset()` on your Datatable see this code in PF 11.0.0

https://github.com/primefaces/primefaces/blob/38c62b7eb8acf8aa5663be43b19e8a14c16f7bae/primefaces/src/main/java/org/primefaces/component/datatable/DataTable.java#L544-L555

**Reply by AlexFTrapp** [reply]
Hi meloware,

thanks for your fast answer. I tried that approach just now. Unfortunelly i was not able to get the result i was lokking for.
1. I tried to call reset() on the clientSide instead of "clearFilters()" but with this, the GUI froze and the devTools in the browser tell me that "TypeError: PF(...).reset is not a function"
2. I tried in the backingBean. Therefore i looked up the datatable using "FacesContext.getCurrentInstance().getViewRoot().findComponent" and called "reset()" on this. But even though the GUI was not updated.  Should there be more than the update-call of the commandButton to force the GUI to refresh the datatable?

Thanks for your help
Alex

**Reply by melloware** [reply]
Yes `reset()` is a server side call to reset the state of the datatable and then you need to `update` your client side datatable after reset.  

Please provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.

**Reply by AlexFTrapp** [reply]
Hi melloware,

thanks again for your help. As far as I can see, the styleclass "ui-helper-hidden" is not removed from the <th>-Element of the columns. That is the only difference I found to when the toggler-function enables/disables the visibility of a column and when i do a reset. 
Greets Alex

**Reply by melloware** [reply]
Have you tried 14.0.6?  I know you are using 11.0.0 but just wondering if this is something that is already fixed.
